src_lockfree/treiber_stack.ml

```diff
@@ -14,17 +14,16 @@ let push q v =
   let new_node = Next (v, head) in
   if Atomic.compare_and_set q.head head new_node then ()
   else
-    let b = Backoff.create () in
-    Backoff.once b;
+    let b = Backoff.create () |> Backoff.once in
```

Comment:
Hmm... I'd recommend considering to rewrite the code here a bit more.  A kind of (recursive) worker- (non-recursive) wrapper idiom:

```ocaml
let rec push backoff q v =
  ...

let push q v = push Backoff.default q v
```

I believe this would make the code a bit shorter / simpler and should perform reasonably well.